### PR TITLE
[#1022] Mock data for course (connects #1022)

### PR DESCRIPTION
**Pull request by Pr0chin**

### CODE REVIEW

**Comment by Pr0chin** [comment]
d1
d2
d3
d4
d5
d6
d7

d8

File: mock_data [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+// Increase Limits so couchapp push works correctly
+function increaseLimits() {
+    exec('sudo launchctl limit maxfiles 150056 150056', function doneWithLaunchCtl() {
+        exec('sudo ulimit -n 150056', function doneWithUlimit() {
+            return createDummyCourses()
+        })
+    })
+}
+
+var n_courses = 1 ; //Number of Iteration to run for courses
```
Comment (Pr0chin) [inline mock_data]: @dogi  as i have initialize variable var n_courses = 1 as a static value we can remove variable and loop statement for course if we need only one course.

**Comment by leomaxi** [comment]
Looks okay, what about quiz ?. Does the scrip also create a simple quiz in the course ?

**Review by nirojdyola — DISMISSED** [review]

**Comment by Pr0chin** [comment]
@leonardmensah  yes :D it creates course having one steps and 4 different types of question i.e MCQ, single-text, attachment and comment-box. if u want to run this mock data then u should run 
**node [script_name] [url]**

**Review by razu9861 — DISMISSED** [review]

**Review by Rupesh87 — DISMISSED** [review]

File: tests/mock_data [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    .parse(process.argv);
+if (!program.args[0]) return console.log('No CouchDB URL provided. Quiting now.')
+var couchUrl = program.args[0]
+var nano = require('nano')(couchUrl)
+
+function start() {
+    increaseLimits()
+}
+
+// Increase Limits so couchapp push works correctly
```
Comment (dogi) [inline tests/mock_data]: not needed on linux please start only when MacOS detected

File: tests/mock_data [hunk 3]
```diff
@@ -0,0 +1,197 @@
+var sys = require('sys')
```
Comment (dogi) [inline tests/mock_data]: please exchange `sys` with `util`

**Review by dogi — CHANGES_REQUESTED** [review]
Generally we want 2 courses:
1 one step and in there all kind of questions
2 make a team meeting adiscuss about how the ultimative Course should look like to later on test stuff on with travis really fast ...

**Comment by Pr0chin** [comment]
aa
a
b
h
d
e
ff
g

**Comment by Pr0chin** [comment]
@dogi I have made mock_data as per your concern. please give me some feedback regarding to course mock_data.
and we ll have team discussion for further course scenario as team is busy doing their stuff.

**Comment by dogi** [comment]
@Rupesh87 @razu9861 @nirojdyola please review again ;)

File: tests/course_mock_data [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    courses.bulk({"docs":defaultCourses}, function(err, res) {
+        if (err)
+            return console.log(err);
+        else {
+            course_id = [];
+            console.log(res.length);
+            for(var i=0;i<res.length;i++){
+                course_id.push(res[i].id);
+            }
+            for(var i = 0; i < 1; i++){
```
Comment (nirojdyola) [inline tests/course_mock_data:80]: If it is only for two courses then it is fine, otherwise it should be change.

**Review by dogi — APPROVED** [review]

**Pull request merged**